[CI] InternalEngineTests.testRestoreLocalHistoryFromTranslog

This one reproduces for me with `./gradlew ':server:test' --tests "org.elasticsearch.index.engine.InternalEngineTests.testRestoreLocalHistoryFromTranslog" -Dtests.seed=9E40590F007B5A62 -Dtests.security.manager=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=da -Dtests.timezone=America/Adak -Dcompiler.java=13`

```
03:27:37   1> [2020-02-16T14:26:51,941][INFO ][o.e.i.e.InternalEngineTests] [testRestoreLocalHistoryFromTranslog] before test
03:27:37   1> [2020-02-16T14:26:52,385][INFO ][o.e.i.e.InternalEngineTests] [testRestoreLocalHistoryFromTranslog] after test
03:27:37   2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.engine.InternalEngineTests.testRestoreLocalHistoryFromTranslog" -Dtests.seed=9E40590F007B5A62 -Dtests.security.manager=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=da -Dtests.timezone=America/Adak -Dcompiler.java=13
03:27:37   2> java.lang.AssertionError: restore from local translog must not add operations to translog
03:27:37     Expected: <23>
03:27:37          but: was <20>
03:27:37         at __randomizedtesting.SeedInfo.seed([9E40590F007B5A62:1BAA42410B03C763]:0)
03:27:37         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
03:27:37         at org.junit.Assert.assertThat(Assert.java:956)
03:27:37         at org.elasticsearch.index.engine.InternalEngineTests.testRestoreLocalHistoryFromTranslog(InternalEngineTests.java:4243)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/907/console

--- a/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -4207,7 +4207,6 @@ public class InternalEngineTests extends EngineTestCase {
             final EngineConfig engineConfig;
             final SeqNoStats prevSeqNoStats;
             final List<DocIdSeqNoAndSource> prevDocs;
-            final List<Translog.Operation> existingTranslog;
             try (InternalEngine engine = createEngine(store, createTempDir(), globalCheckpoint::get)) {
                 engineConfig = engine.config();
                 for (final long seqNo : seqNos) {
@@ -4226,24 +4225,17 @@ public class InternalEngineTests extends EngineTestCase {
                 engine.syncTranslog();
                 prevSeqNoStats = engine.getSeqNoStats(globalCheckpoint.get());
                 prevDocs = getDocIds(engine, true);
-                try (Translog.Snapshot snapshot = engine.getTranslog().newSnapshot()) {
-                    existingTranslog = TestTranslog.drainSnapshot(snapshot, false);
-                }
             }
             try (InternalEngine engine = new InternalEngine(engineConfig)) {
-                final Translog.TranslogGeneration currrentTranslogGeneration = new Translog.TranslogGeneration(
-                    engine.getTranslog().getTranslogUUID(), engine.getTranslog().currentFileGeneration());
+                final long currentTranslogGeneration = engine.getTranslog().currentFileGeneration();
                 engine.recoverFromTranslog(translogHandler, globalCheckpoint.get());
                 engine.restoreLocalHistoryFromTranslog(translogHandler);
                 assertThat(getDocIds(engine, true), equalTo(prevDocs));
                 SeqNoStats seqNoStats = engine.getSeqNoStats(globalCheckpoint.get());
                 assertThat(seqNoStats.getLocalCheckpoint(), equalTo(prevSeqNoStats.getLocalCheckpoint()));
                 assertThat(seqNoStats.getMaxSeqNo(), equalTo(prevSeqNoStats.getMaxSeqNo()));
-                try (Translog.Snapshot snapshot = engine.getTranslog().newSnapshot()) {
-                    assertThat("restore from local translog must not add operations to translog",
-                        snapshot.totalOperations(), equalTo(existingTranslog.size()));
-                    assertThat(TestTranslog.drainSnapshot(snapshot, false), equalTo(existingTranslog));
-                }
+                assertThat("restore from local translog must not add operations to translog",
+                    engine.getTranslog().totalOperationsByMinGen(currentTranslogGeneration), equalTo(0));
             }
             assertConsistentHistoryBetweenTranslogAndLuceneIndex(engine, createMapperService());
         }

